map/bookmark_manager.cpp

```diff
@@ -100,27 +137,152 @@ void BookmarkManager::LoadBookmarks()
 {
   ClearCategories();
 
-  std::string const dir = GetPlatform().SettingsDir();
+  StartAsyncLoading();
+  GetPlatform().RunTask(Platform::Thread::File, [this]()
+  {
+    std::string const dir = GetPlatform().SettingsDir();
+    Platform::FilesList files;
+    Platform::GetFilesByExt(dir, BOOKMARKS_FILE_EXTENSION, files);
+
+    auto collection = std::make_shared<CategoriesCollection>();
+    for (auto const & file : files)
+    {
+      auto cat = BookmarkCategory::CreateFromKMLFile(dir + file);
+      if (m_needTeardown)
+        return;
 
-  Platform::FilesList files;
-  Platform::GetFilesByExt(dir, BOOKMARKS_FILE_EXTENSION, files);
-  for (auto const & file : files)
-    LoadBookmark(dir + file);
+      if (cat != nullptr)
+        collection->emplace_back(std::move(cat));
+    }
+    FinishAsyncLoading(std::move(collection));
+  });
 
   LoadState();
 }
 
-void BookmarkManager::LoadBookmark(std::string const & filePath)
+void BookmarkManager::LoadBookmark(std::string const & filePath, bool isTemporaryFile)
 {
-  auto cat = BookmarkCategory::CreateFromKMLFile(filePath);
-  if (cat != nullptr)
+  StartAsyncLoading();
```

Comment:
Am I right that StartAsyncLoading() do not load anything, but triggers callbacks only? If so, could you please rename the method?